spec/lib/diaspora_federation/entities/reshare_spec.rb

```diff
@@ -74,6 +74,31 @@ module DiasporaFederation
             Entities::Reshare.from_xml(Nokogiri::XML(xml).root)
           }.to raise_error Entity::ValidationError
         end
+
+        it "validates a reshare with no root" do
+          data[:root_author] = nil
+          data[:root_guid] = nil
+
+          expect {
+            Entities::Reshare.from_xml(Nokogiri::XML(xml).root)
+          }.not_to raise_error
```

Comment:
Instead of `not_to raise_error` I would rather check if the parsed `root_author` and `root_guid` is `nil`. The "not to raise error"-part is implicit then.